### PR TITLE
Product listing page

### PR DESCRIPTION
**Pull request by KoushikKintali**
Migrated the HTML screens to React
Populated products from fetchProductsApi
Added ADD_TO_CART functionality
Added ADD_TO_WISHLIST functionality
Added REMOVE_FROM_WISHLIST functionality

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
 import "./App.css";
+import Navbar from './components/navbar/Navbar';
+import ProductListing from './pages/ProductListing/ProductListing';
```
Comment (MiheerTamkhane) [inline src/App.js:3]: Use barrel exports here.

File: src/App.js [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
-  return (
-    <div className="App">
-        Home Page
-    </div>
-  );
+
+    return (
+        <div className="App">
+            <Navbar></Navbar>
+            <ProductListing></ProductListing>
```
Comment (MiheerTamkhane) [inline src/App.js]: You can just use self closing tag here : <Navbar />

File: src/components/filter/Filter.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                <div className="slider-component">
+                    <span className="slider-value-min" id="silder-min-value">0</span>
+                    <input type="range" min="0" max="100" id="slider" className="slider" defaultValue={0} />
+                    <span className="slider-value-max" id="silder-max-value">100</span>
+                </div>
+            </div>
+            <div className="category-group">
+                <p className="title">Category</p>
+                <ul className="category-group-list">
+                    <li><input type="checkbox" /> Category1</li>
```
Comment (MiheerTamkhane) [inline src/components/filter/Filter.jsx]: Wrap the inputs in <label> tag it would be good practice.

File: src/components/filter/Filter.jsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                        <span className="material-icons-outlined rating-element">star</span>
+                    </div>
+                </div>
+            </div>
+            <div className="sort-group">
+                <p className="title">Sort by</p>
+                <div className="sort-group-list">
+                    <div>
+                        <input type="radio" /> Price - Low to High
+                    </div>
```
Comment (MiheerTamkhane) [inline src/components/filter/Filter.jsx]: here also, instead of div wrap inside label

File: src/components/navbar/Navbar.jsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import logo from '../../assets/logo.png';
+import { useCart } from '../../context/cart-context';
+
+
+const Navbar = () => {
+
+    const { store } = useCart();
+    console.log('store.itemsInWishlist', store.itemsInWishlist)
```
Comment (MiheerTamkhane) [inline src/components/navbar/Navbar.jsx]: Remove consoles also after checks.

File: src/components/product/Product.jsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    const getDiscountedPrice = (price, discount) => {
+        return Number(price - ((price * discount) / 100));
+    }
+
+    const isItemInCart = (id) => {
+        const { itemsInCart } = store;
+        return itemsInCart.includes(id);
+    }
+
+    const isItemInWishlist = (id) => {
```
Comment (MiheerTamkhane) [inline src/components/product/Product.jsx:18]: I liked how you have created this utility function. Try to move it in diff folder the code will be more clean

File: src/pages/ProductListing/ProductListing.jsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+import './ProductListing.css';
+
+const ProductListing = () => {
+
+    const [products, setProducts] = useState([]);
+
+    const fetchProducts = async () => {
+        try {
+            const products = await axios.get('/api/products');
+            setProducts([...products.data.products]);
```
Comment (MiheerTamkhane) [inline src/pages/ProductListing/ProductListing.jsx:15]: Is there any specific reason to destructure the products and the passing to state, coz you are getting already an array of products? So you can directly put them in.

File: src/pages/ProductListing/ProductListing.jsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        }
+    };
+
+    useEffect(() => fetchProducts(), []);
+
+
+    return (
+        <>
+            <section className="content-body products-body">
+                <Filter></Filter>
```
Comment (MiheerTamkhane) [inline src/pages/ProductListing/ProductListing.jsx]: you can use self closing tag here also

**Review by MiheerTamkhane — APPROVED** [review]
The code is readable and looks nice. Just look into the comments I've write, Other than that all good

File: src/pages/ProductListing/ProductListing.jsx (hunk 7 above)
Comment (KoushikKintali) [reply src/pages/ProductListing/ProductListing.jsx:15]: No, there is no such reason, we can use either ways

**Review by SomeEarth86 — APPROVED** [review]

**Review by SomeEarth86 — COMMENTED** [review]
Looks good to merge.

**Review by raghav6202 — APPROVED** [review]
Code looks clean.Look into the changes sugggested by Miheer . Good to merge👍

**Pull request merged**